Nodejs/Product/TypeScriptSourceMapReader/SourceMapReader.cs

```diff
@@ -0,0 +1,290 @@
+﻿//*********************************************************//
+//    Copyright (c) Microsoft. All rights reserved.
+//    
+//    Apache 2.0 License
+//    
+//    You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//    
+//    Unless required by applicable law or agreed to in writing, software 
+//    distributed under the License is distributed on an "AS IS" BASIS, 
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+
+using Newtonsoft.Json;
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.NodejsTools.TypeScriptSourceMapReader {
+    /// <summary>
+    /// Helper class to read the source map and get the decoded source map
+    /// It also has some apis to test if the url is js url and get the sourceMap url
+    /// </summary>
+    public class SourceMapReader : SourceMapTextReader {
```

Comment:
Both of `SourceMapReader` and `SourceMapTextReader`  should be `static` since, as far as I can tell, neither has any state or needs inheritance.